Std/Data/String.lean

```diff
@@ -35,9 +35,66 @@ termination_by go start stop => stop.byteIdx + 4 - start.byteIdx
 
 end Substring
 
-
 namespace String
 
+@[simp]
+theorem Pos.byteIdx_zero : (0 : Pos).byteIdx = 0 :=
+  rfl
+
+@[simp]
+theorem Pos.mk_zero : ⟨0⟩ = (0 : Pos) :=
+  rfl
+
+@[simp]
+theorem Pos.eq_iff {i₁ i₂ : Pos} : i₁ = i₂ ↔ i₁.byteIdx = i₂.byteIdx :=
+  ⟨fun h ↦ h ▸ rfl, fun h ↦ show ⟨i₁.byteIdx⟩ = (⟨i₂.byteIdx⟩ : Pos) from h ▸ rfl⟩
+
+@[simp]
+theorem Pos.add_eq (p₁ p₂ : Pos) : p₁ + p₂ = ⟨p₁.byteIdx + p₂.byteIdx⟩ :=
+  rfl
+
+@[simp]
+theorem Pos.sub_eq (p₁ p₂ : Pos) : p₁ - p₂ = ⟨p₁.byteIdx - p₂.byteIdx⟩ :=
+  rfl
+
+@[simp]
+theorem Pos.add_char (p : Pos) (c : Char) : p + c = ⟨p.byteIdx + csize c⟩ :=
+  rfl
+
+@[simp]
+theorem Pos.zero_add_char (c : Char) : (0 : Pos) + c = ⟨csize c⟩ := by
+  simp
+
+@[simp]
+theorem Pos.add_string (p : Pos) (s : String) : p + s = ⟨p.byteIdx + s.utf8ByteSize⟩ :=
+  rfl
+
+@[simp]
+theorem Pos.zero_add_string (s : String) : (0 : Pos) + s = ⟨s.utf8ByteSize⟩ := by
+  simp
+
+@[simp]
+theorem Pos.le_iff {i₁ i₂ : Pos} : i₁ ≤ i₂ ↔ i₁.byteIdx ≤ i₂.byteIdx :=
+  ⟨id, id⟩
+
+@[simp]
+theorem Pos.lt_iff {i₁ i₂ : Pos} : i₁ < i₂ ↔ i₁.byteIdx < i₂.byteIdx :=
```

Comment:
Lemmas go in a `Lemmas` file unless they are absolutely necessary for defining functions.

Reply:
I'll do it after I wake up.